node/impl/full/eth.go

```diff
@@ -633,19 +637,23 @@ func (a *EthModule) EthFeeHistory(ctx context.Context, blkCount ethtypes.EthUint
 	}
 
 	// Reverse the arrays; we collected them newest to oldest; the client expects oldest to newest.
-
 	for i, j := 0, len(baseFeeArray)-1; i < j; i, j = i+1, j-1 {
 		baseFeeArray[i], baseFeeArray[j] = baseFeeArray[j], baseFeeArray[i]
 	}
 	for i, j := 0, len(gasUsedRatioArray)-1; i < j; i, j = i+1, j-1 {
 		gasUsedRatioArray[i], gasUsedRatioArray[j] = gasUsedRatioArray[j], gasUsedRatioArray[i]
 	}
 
-	return ethtypes.EthFeeHistory{
+	ret := ethtypes.EthFeeHistory{
 		OldestBlock:   ethtypes.EthUint64(oldestBlkHeight),
 		BaseFeePerGas: baseFeeArray,
 		GasUsedRatio:  gasUsedRatioArray,
-	}, nil
+	}
+	if params.RewardPercentiles != nil {
+		reward := make([][]ethtypes.EthBigInt, 0)
+		ret.Reward = &reward
+	}
```

Comment:
Is this a todo, or is there actually a valid reason why we always populate ret.Reward with an empty array?

Reply:
This is a todo that we should probably implement in the future.

Reply:
This is a todo that we should probably implement in the future.

Reply:
So this should return:

> An array of effective priority fee per gas data points from a single block.

We need to calculate the requested percentiles of _effective_ gas premium based on the newest block (I presume it's the newest, need to dig in). _Effective_ means we're clamped at the gas_fee_cap - base_fee.

Reply:
https://github.com/filecoin-project/lotus/issues/10236